### PR TITLE
Add common field type linter

### PR DESCRIPTION
**Pull request by mbryzek**

### CODE REVIEW

File: src/main/scala/io/flow/lint/linters/CommonFieldTypes.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+/**
+  * For well known field names, enforce specific types to ensure
+  * consistency. For example, all fields named 'id' or 'name' must be
+  * strings.
+  */
+case object CommonFieldTypes extends Linter with Helpers {
+
+  val Expected = Map(
+    "id" -> "string",
+    "name" -> "string"
```
Comment (chrhicks) [inline src/main/scala/io/flow/lint/linters/CommonFieldTypes.scala]: `guid -> uuid`
`email -> string` (stretch)

??

**Comment by mikeclimbrock** [comment]
tests passed, lgtm

**Comment by chrhicks** [comment]
:+1:

File: src/main/scala/io/flow/lint/linters/CommonFieldTypes.scala (hunk 1 above)
Comment (mbryzek) [inline src/main/scala/io/flow/lint/linters/CommonFieldTypes.scala]: added

**Pull request merged**